sdk/program/src/vote/state/mod.rs

```diff
@@ -414,13 +424,17 @@ impl VoteState {
             }
         }
 
-        self.epoch_credits.last_mut().unwrap().1 += credits;
+        self.epoch_credits.last_mut().unwrap().1 =
+            self.epoch_credits.last().unwrap().1.saturating_add(credits);
     }
 
-    #[allow(clippy::integer_arithmetic)]
     pub fn nth_recent_vote(&self, position: usize) -> Option<&Lockout> {
         if position < self.votes.len() {
-            let pos = self.votes.len() - 1 - position;
+            let pos = self
+                .votes
+                .len()
+                .checked_sub(position)
```

Comment:
can this just be simplified to `checked_sub(position + 1)`

Reply:
clippy's not smart enough to see the `if` so it considers `position + 1` as integer arithmetic

Reply:
ah yeah duh